src/networking.c

```diff
@@ -2878,6 +2934,47 @@ int clientSetNameOrReply(client *c, robj *name) {
     return result;
 }
 
+/* This function implements CLIENT SETMETA, including replying to the
+ * user with an error if the charset is wrong (in that case C_ERR is
+ * returned). If the function succeeded C_OK is returned, and it's up
+ * to the caller to send a reply if needed.
+ *
+ * Setting an empty string as meta has the effect of unsetting the
+ * currently set meta: the client will remain without meta.*/
+int clientSetMetaOrReply(client *c)
+{
+    int len = c->argc;
+
+    if (c->meta == NULL)
+    {
+        c->meta = dictCreate(&hashDictType);
+    }
+
+    /* Setting the client meta do not have params actually removes
+     * the current meta. */
+    if (len == 2)
+    {
+        dictRelease(c->meta);
+        c->meta = dictCreate(&hashDictType);
+    
+        return C_OK;
+    }
+
+    int i;
+    for (i = 0; i < (len - 2)/2; i++) {
+        robj *o1 = c->argv[2+i*2];
+        sds key = sdsdup(o1->ptr);
+        robj *o2 = c->argv[3+i*2];
+        sds val = sdsdup(o2->ptr);
```

Comment:
Personal preference, but a bit more readable:

```suggestion
    for (int i = 2; i < c->argc; i++) {
        sds key = sdsdup(c->argv[i++]);
        sds val = sdsdup(c->argv[i]);
        ....    
    }
```

This is similar to what you did earlier.